tests/gw/test_waveform_dataset.py

```diff
@@ -50,80 +54,69 @@
 
 
 @pytest.fixture
-def generate_waveform_dataset_small(venv_dir='venv'):
+def temp_dir() -> Generator[Path, None, None]:
+    """
+    Fixture to provide a temporary directory path using pathlib.Path.
+
+    :return: A pathlib.Path object pointing to the temporary directory.
+    """
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        yield Path(tmpdirname)
+
+
+@pytest.fixture
+def generate_waveform_dataset_small(temp_dir: Path) -> Path:
```

Comment:
- calls the new function instead of the executable
- uses the tempfile package to create a temporary folder